### ISSUE TITLE
model.prediction does not match model.evaluation loss error

### ISSUE THREAD

**Issue by Manal4** [Hvass-Labs/TensorFlow-Tutorials#119]
Hi
Thanks for the tutorial.
I applied the same code using different data, it worked perfectly, I added the mes, mae metrics in addition to the loss fuction,  but when I tried to calculate the individual MSEs, it seems that the error from model.prediction() does not match the error from the model.evaluation().
I just add this line in the comparison function:

                                 `  mean_squared_error(signal_true,signal_pred) `
I also divide it by 1000 but the total of mean squared errors from different signals does not match the mse from the model.evaluation().
thanks.

**Comment by Manal4** [comment]
I saw @bstriner [https://github.com/bstriner](url)  reply regarding a similar issue here [https://github.com/keras-team/keras/issues/5140](url), but I couldn't fix it.

**Comment by Manal4** [comment]
I saw @bstriner  reply #https://github.com/keras-team/keras/issues/5140#issuecomment-274554133  regarding a similar issue in #https://github.com/keras-team/keras/issues/5140, but I couldn't fix it, it seems to me that shapes/data types are correct.

**Comment by Hvass-Labs** [comment]
I don't know what you are referring to, but I have just updated many of the tutorials to support TensorFlow 2, so you can try again with the new version.

**Comment by STEVE-ELF** [comment]
Thanks for your letters.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Hvass-Labs"<notifications@github.com&gt;;
发送时间:&nbsp;2020年4月2日(星期四) 晚上8:52
收件人:&nbsp;"Hvass-Labs/TensorFlow-Tutorials"<TensorFlow-Tutorials@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;Re: [Hvass-Labs/TensorFlow-Tutorials] model.prediction does not match model.evaluation loss error (#119)





 
I don't know what you are referring to, but I have just updated many of the tutorials to support TensorFlow 2, so you can try again with the new version.
